src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TemplateRenderer.cs

```diff
@@ -82,6 +84,8 @@ internal class TemplateRenderer
             _viewData = viewData;
             _templateName = templateName;
             _readOnly = readOnly;
+            _logger = _viewContext.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>()
+                .CreateLogger<TemplateRenderer>();
```

Comment:
I'm not sure we should log inside the TemplateRender-er. All of the places we're doing logger elsewhere in the PR are where the user directly asked for a view.

If we really want to log **all** view discovery, then we can put that in the view engine.